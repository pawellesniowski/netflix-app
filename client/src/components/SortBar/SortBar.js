import React from 'react';
import './SortBar.scss';

const SortBar = ({moviesNumber=0}) => {
  return (
    <div className="sort-bar" data-cy="sort-bar">
      <span className="sort-bar-info">{moviesNumber} movies found</span>
      <div className="sort-bar-menu">
        <span>Sort by</span>
        <ul className="sort-bar-menu-options">
          <li key="release-date" className="sort-bar-menu-item">release date</li>
          <li key="rating" className="sort-bar-menu-item">rating</li>
        </ul>
      </div>
    </div>
  );
};

export default SortBar;
