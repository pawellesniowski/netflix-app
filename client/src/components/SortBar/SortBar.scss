@import "../../common/styles/colors";
@import "../../common/styles/mixins.scss";

.sort-bar {
  @include contentContainer;
  background-color: $color-grey-mid-grey;
  color: $color-grey-black;
  font-size: 1.5rem;
  display: flex;
  justify-content: space-between;

  .sort-bar-menu {
    display: flex;

    .sort-bar-menu-options {
      list-style: none;
      display: flex;

      .sort-bar-menu-item {
        margin-left: 2rem;
      }
    }
  }
}

