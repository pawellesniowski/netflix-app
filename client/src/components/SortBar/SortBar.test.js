import React from 'react';
import {shallow} from "enzyme/build";
import SortBar from "./SortBar";

describe('SortBar Component', () => {
    let component;
    beforeEach(() => {
        component = shallow(<SortBar/>);
    });

    it("Should have container", () => {
        expect(component.find('[data-cy="sort-bar"]')).toHaveLength(1);
    });

    test("Snapshot test", () => {
        expect(component).toMatchSnapshot();
    });
});

describe('SortBar Component with props', () => {
    it('Shows message "0 movies found" if no props send', () => {
        const component = shallow(<SortBar />);
        expect(component.text()).toMatch("0 movies foundSort");
    });

    it('Shows message "3 movies found" if no props send', () => {
        const component = shallow(<SortBar moviesNumber="3"/>);
        expect(component.text()).toMatch("3 movies foundSort");
    });
});

