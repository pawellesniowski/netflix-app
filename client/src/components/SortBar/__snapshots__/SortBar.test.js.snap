// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar Component Snapshot test 1`] = `ShallowWrapper {}`;
